Lib/concurrent/futures/_base.py

```diff
@@ -520,7 +521,7 @@ def submit(self, fn, *args, **kwargs):
         """
         raise NotImplementedError()
 
-    def map(self, fn, *iterables, timeout=None, chunksize=1):
+    def map(self, fn, *iterables, timeout=None, chunksize=1, prefetch=None):
         """Returns an iterator equivalent to map(fn, iter).
```

Comment:
Ah actually... I am wrong here as well :) The interaction between `chunksize` and `prefetch` is not what I thought. What happens is that you actually prefetch the number of items equal to `(_max_workers + prefetch) * chunksize`; and contrary to what I said, `chunksize` does *not* prevent infinite (or very large) generators from overflowing the memory.

So I think it's ok to have your PR handle both `ProcessPoolExecutor` and `ThreadPoolExecutor`, but you probably want to adjust `prefetch` to account for `chunksize`, for example by setting `prefetch = max(1, prefetch // chunksize)` and avoiding increasing it by another `_max_workers` in `map`.